gbpservice/nfp/orchestrator/config_drivers/heat_driver.py

```diff
@@ -820,6 +820,105 @@ def _create_node_config_data(self, auth_token, tenant_id,
                     'description'] = str(common_desc)
 
                 nf_desc = str(firewall_desc)
+        elif service_type == pconst.VPN:
+            # rvpn_l3_policy = self._get_rvpn_l3_policy(auth_token,
+            #    provider, update)
+            # if rvpn_l3_policy is None:
+            #    return None, None
+            # config_param_values['ClientAddressPoolCidr'] = rvpn_l3_policy[
+            #    'ip_pool']
+            config_param_values['Subnet'] = (
+                consumer_port['fixed_ips'][0]['subnet_id']
+                if consumer_port else None)
+            l2p = self.gbp_client.get_l2_policy(
+                auth_token, provider['l2_policy_id'])
+            l3p = self.gbp_client.get_l3_policy(
+                auth_token, l2p['l3_policy_id'])
+            config_param_values['RouterId'] = l3p['routers'][0]
+            stitching_cidr = service_details['consumer_subnet']['cidr']
+            mgmt_gw_ip = self._get_management_gw_ip(auth_token)
+            if not mgmt_gw_ip:
+                return None, None
+
+            services_nsp = self.gbp_client.get_network_service_policies(
+                auth_token,
+                filters={'name': ['nfp_services_nsp']})
+            if not services_nsp:
+                fip_nsp = {
+                    'network_service_policy': {
+                        'name': 'nfp_services_nsp',
+                        'description': 'nfp_implicit_resource',
+                        'shared': False,
+                        'tenant_id': tenant_id,
+                        'network_service_params': [
+                            {"type": "ip_pool", "value": "nat_pool",
+                             "name": "vpn_svc_external_access"}]
+                    }
+                }
+                nsp = self.gbp_client.create_network_service_policy(
+                    auth_token, fip_nsp)
+            else:
+                nsp = services_nsp[0]
+            if not base_mode_support:
+                stitching_pts = self.gbp_client.get_policy_targets(
+                    auth_token,
+                    filters={'port_id': [consumer_port['id']]})
+                if not stitching_pts:
+                    LOG.error(_LE("Policy target is not created for the "
+                                  "stitching port"))
+                    return None, None
+                stitching_ptg_id = (
+                    stitching_pts[0]['policy_target_group_id'])
+            else:
+                stitching_ptg_id = consumer['id']
+            self.gbp_client.update_policy_target_group(
+                auth_token, stitching_ptg_id,
+                {'policy_target_group': {
+                    'network_service_policy_id': nsp['id']}})
+            if not base_mode_support:
+                floatingips = self.neutron_client.get_floating_ips(
```

Comment:
Lets check if all the gets we do here can be avoided by checking if the data is already available in the local variables.